### ISSUE TITLE
Array literals on multiple lines are incorrectly reported as errors

### ISSUE THREAD

**Issue by avikivity** [redhat-plumbers-in-action/differential-shellcheck#470]
### Type of issue

Bug Report

### Description

When shellcheck sees an array literal assigned to a variable, with the array literal spread over multiple lines, it complains.

Example report:

```
Error: SHELLCHECK_WARNING:
./install-dependencies.sh:164:22: warning[SC2192]: This array element has no value. Remove spaces after = or use "" for empty string.
#  161|       [pykmip]=
#  162|   )
#  163|   
#  164|-> pip_symlinks=(
#  165|       scylla-api-client
#  166|   )
#  167|   
```

pip_symlinks is a perfectly valid array with one happy element.

Seen in https://github.com/scylladb/scylladb/pull/22497.

### Describe the solution you'd like

shellcheck should recognize that an unbalanced parentheses means it should continue parsing until the balance is achieved.

**Comment by jamacku** [comment]
Thank you for reaching out. I have tried to reproduce your issue in this PR:
- https://github.com/jamacku/scylladb/pull/1

The warnings reported by ShellCheck are correct, and GitHub correctly displays them. 

![Image](https://github.com/user-attachments/assets/88be0899-ced3-475d-bd56-e211094c0142)

The problem is that the lines are shifted in the GitHub Action logs (the `pip_symlinks` array is the correct shell). A high concentration (lines 158-165) of SC2192 might cause this. I would suggest you to fix [SC2192](https://www.shellcheck.net/wiki/SC2192) by adding `""`

**Comment by avikivity** [comment]
Ah, thanks. Is there a bug filed for the incorrect source line attribution?

**Comment by jamacku** [comment]
No, there is no tracker for this issue. I'm trying to find out what has happened there.

**Comment by jamacku** [comment]
@avikivity I have debugged and fixed the issue. The problem was that git-lfs was absent in the Differential ShellCheck image.

```
This repository is configured for Git LFS but 'git-lfs' was not found on your path. If you no longer wish to use Git LFS, remove this hook by deleting the 'post-checkout' file in the hooks directory (set by 'core.hookspath'; usually '.git/hooks').
```

This caused defects to be mapped against an old file, and lines were shifted to the wrong place.

It should be fixed by https://github.com/redhat-plumbers-in-action/differential-shellcheck/commit/7aba0c12baa4dd19c4e0ce32934308014f840a9a.

**Comment by jamacku** [comment]
The issue should be fixed in the latest release https://github.com/redhat-plumbers-in-action/differential-shellcheck/releases/tag/v5.5.1

@avikivity Feel free to reopen if the issue persists.

**Comment by avikivity** [comment]
Thanks a lot.
